table/block_based/block_based_table_builder.cc

```diff
@@ -1470,30 +1481,15 @@ Status BlockBasedTableBuilder::InsertBlockInCompressedCache(
   return s;
 }
 
-Status BlockBasedTableBuilder::InsertBlockInCacheHelper(
+template <typename TBlocklike>
+Status BlockBasedTableBuilder::InsertBlockInCache(
     const Slice& block_contents, const BlockHandle* handle,
-    BlockType block_type) {
-  Status s;
-  if (block_type == BlockType::kData || block_type == BlockType::kIndex) {
```

Comment:
I understand the motivation to get rid of this ugly mess, but (as above) we should try for a minimal fix for now for confidence in back-porting.